watertap/unit_models/electroNP_ZO.py

```diff
@@ -0,0 +1,494 @@
+
+        init_log.info_high(
+            "Initialization Step 2 {}.".format(idaeslog.condition(results))
+        )
+
+        # ---------------------------------------------------------------------
+        # Release Inlet state
+        self.properties_in.release_state(flags, outlvl)
+
+        init_log.info("Initialization Complete: {}".format(idaeslog.condition(results)))
+
+        if not check_optimal_termination(results):
+            raise InitializationError(
+                f"{self.name} failed to initialize successfully. Please check "
+                f"the output logs for more information."
+            )
+
+    def _get_performance_contents(self, time_point=0):
+        var_dict = {}
+        var_dict["Water Recovery"] = self.recovery_frac_mass_H2O[time_point]
+        for j in self.config.property_package.solute_set:
+            var_dict[f"Solute Removal {j}"] = self.removal_frac_mass_comp[time_point, j]
+        var_dict["Electricity Demand"] = self.electricity[time_point]
+        var_dict["Electricity Intensity"] = self.energy_electric_flow_mass
+        var_dict[
+            "Dosage of magnesium chloride per treated phosphorus"
+        ] = self.magnesium_chloride_dosage
+        var_dict["Magnesium Chloride Demand"] = self.MgCl2_flowrate[time_point]
+        return {"vars": var_dict}
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Inlet": self.inlet,
+                "Treated": self.treated,
+                "Byproduct": self.byproduct,
+            },
+            time_point=time_point,
+        )
+
+    def calculate_scaling_factors(self):
+        # Get default scale factors and do calculations from base classes
+        for t, v in self.water_recovery_equation.items():
+            iscale.constraint_scaling_transform(
+                v,
+                iscale.get_scaling_factor(
+                    self.properties_in[t].get_material_flow_terms("Liq", "H2O"),
+                    default=1,
+                    warning=True,
+                    hint=" for water recovery",
+                ),
+            )
+
+        for t, v in self.water_balance.items():
+            iscale.constraint_scaling_transform(
+                v,
+                iscale.get_scaling_factor(
+                    self.properties_in[t].get_material_flow_terms("Liq", "H2O"),
+                    default=1,
+                    warning=False,
+                ),
+            )  # would just be a duplicate of above
+
+        for (t, p, j), v in self.solute_removal_equation.items():
+            iscale.constraint_scaling_transform(
+                v,
+                iscale.get_scaling_factor(
+                    self.properties_in[t].get_material_flow_terms(p, j),
+                    default=1,
+                    warning=True,
+                    hint=" for solute removal",
+                ),
```

Comment:
might not need these scaling transforms if we fix the variables and not build as constraints.

Reply:
I still keep the variables for now.